@xen-orchestra/backups/writers/_DeltaBackupWriter.js

```diff
@@ -7,21 +7,21 @@ const { chainVhd, checkVhdChain, default: Vhd } = require('vhd-lib')
 const { createLogger } = require('@xen-orchestra/log')
 const { dirname } = require('path')
 
+const { formatFilenameDate } = require('../_filenameDate')
+const { getOldEntries } = require('../_getOldEntries')
+const { getVmBackupDir } = require('../_getVmBackupDir')
+const { Task } = require('../Task')
+
+const { MixinBackupWriter } = require('./_MixinBackupWriter')
+const { AbstractDeltaWriter } = require('./_AbstractDeltaWriter')
 const { checkVhd } = require('./_checkVhd')
-const { formatFilenameDate } = require('./_filenameDate')
-const { getOldEntries } = require('./_getOldEntries')
-const { getVmBackupDir } = require('./_getVmBackupDir')
 const { packUuid } = require('./_packUuid')
-const { Task } = require('./Task')
 
 const { warn } = createLogger('xo:backups:DeltaBackupWriter')
 
-exports.DeltaBackupWriter = class DeltaBackupWriter {
+exports.DeltaBackupWriter = class DeltaBackupWriter extends MixinBackupWriter(AbstractDeltaWriter) {
   constructor(backup, remoteId, settings) {
```

Comment:
Pass an object directly so you can remove implementations' constructors entirely.